test/TensorFlowRuntime/sese_regions.swift

```diff
@@ -0,0 +1,159 @@
+  } catch { // expected-note {{value used here}}
+    // expected-warning @+2 {{value implicitly copied to the host}}
+    // expected-warning @+1 {{value implicitly copied to the host}}
+    result += 5
+  }
+  return result
+}
+
+// expected-note @+3 {{value used here}}
+// expected-note @+2 {{value used here}}
+// expected-note @+1 {{value used here}}
+SESERegionTests.testAllBackends("testFunctionRegion") { 
+  expectEqualWithScalarTensor(1, testFunctionRegion(99))
+  expectEqualWithScalarTensor(7, testFunctionRegion(101))
+  expectEqualWithScalarTensor(3, testFunctionRegion(-99))
+  expectEqualWithScalarTensor(9, testFunctionRegion(-101))
+}
+
+// CHECK-LABEL:--- XLA CFG Canonicalize: {{.*}}testFunctionRegion{{.*}}
+// CHECK:[sequence
+// CHECK:  {condition Header: {{bb[0-9]+}}
+// CHECK:    {condition Header: {{bb[0-9]+}}
+// CHECK:      block {{bb[0-9]+}}
+// CHECK:      [sequence
+// CHECK:        block {{bb[0-9]+}}
+// CHECK:        {function
+// CHECK:          block [[FBLK:bb[0-9]+]]
+// CHECK:        }]}
+// CHECK:    {condition Header: {{bb[0-9]+}}
+// CHECK:      block {{bb[0-9]+}}
+// CHECK:      [sequence
+// CHECK:        block {{bb[0-9]+}}
+// CHECK:        {function
+// CHECK:          block [[FBLK]]
+// CHECK:        }]}}
+// CHECK:  block {{bb[0-9]+}}]
+// CHECK:--- XLA CFG Canonicalize end
+
+public func testFunctionRegionWithLoop(_ count : Int32) -> Tensor<Int32> {
+  var result = Tensor<Int32>(0)
+  do {
+    if count > 0 {
+      if count < 100 {
+        // expected-warning @+2 {{value implicitly copied to the host}}
+        // expected-warning @+1 {{value implicitly copied to the host}}
+        result += 1
+      } else {
+        result += 2
+        throw TestError.PositiveOutOfBound
+      }
+    } else {
+      if count > -100 {
+        // expected-warning @+2 {{value implicitly copied to the host}}
+        // expected-warning @+1 {{value implicitly copied to the host}}
+        result += 3
+      } else {
+        result += 4
+        throw TestError.NegativeOutOfBound
+      }
+    } // expected-note {{value used here}}
+    // expected-note @+1 {{value used here}}
+  } catch { // expected-note {{value used here}}
+    var i: Int32 = 0
+    while i < 2 {
+      // expected-warning @+2 {{value implicitly copied to the host}}
+      // expected-warning @+1 {{value implicitly copied to the host}}
+      result += 5
+      i += 1
+    }
+  }
+  return result
+}
+
+// expected-note @+3 {{value used here}}
+// expected-note @+2 {{value used here}}
+// expected-note @+1 {{value used here}}
+SESERegionTests.testAllBackends("testFunctionRegionWithLoop") { 
+  expectEqualWithScalarTensor(1, testFunctionRegionWithLoop(99))
+  expectEqualWithScalarTensor(12, testFunctionRegionWithLoop(101))
+  expectEqualWithScalarTensor(3, testFunctionRegionWithLoop(-99))
+  expectEqualWithScalarTensor(14, testFunctionRegionWithLoop(-101))
+}
+
+
+// CHECK-LABEL:--- XLA CFG Canonicalize: {{.*}}testFunctionRegionWithLoop{{.*}}
```

Comment:
it's a good idea to do some compiler testing on this runtime test!

do we want to also confirm we are duplicating some graph ops when we lower a shared SIL BB to graph def?

also, the CFG below is quite detailed -- is it worth abstracting it out a bit, and only check on the essences?